File: policysquare/lib/data/models/claim_story.dart
import 'package:json_annotation/json_annotation.dart';

part 'claim_story.g.dart';

@JsonSerializable()
class ClaimStory {
  String? id;
  String title;
  String? category;
  String caseName;
  String? court;
  String? issue;
  String story;
  String? verdict;
  List<String>? principles;
  String? bottomLine;
  String? imagePath;
  String? createdAt;

  ClaimStory({
    this.id,
    required this.title,
    this.category,
    required this.caseName,
    this.court,
    this.issue,
    required this.story,
    this.verdict,
    this.principles,
    this.bottomLine,
    this.imagePath,
    this.createdAt,
  });

  factory ClaimStory.fromJson(Map<String, dynamic> json) =>
      _$ClaimStoryFromJson(json);
  Map<String, dynamic> toJson() => _$ClaimStoryToJson(this);
}

File: backend/src/main/java/com/policysquare/commercial/model/ClaimStory.java
package com.policysquare.commercial.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@Table(name = "claim_stories")
public class ClaimStory {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    private String title;
    
    private String category; // e.g. Commercial, Health, Life, Motor
    
    @Column(name = "case_name")
    private String caseName;
    
    private String court;
    
    @Column(columnDefinition = "TEXT")
    private String issue;

    @Column(columnDefinition = "TEXT")
    private String story; // The Claim Story

    @Column(columnDefinition = "TEXT")
    private String verdict;

    @ElementCollection
    private List<String> principles; // Critical Legal Principles

    @Column(columnDefinition = "TEXT")
    private String bottomLine;

    private String imagePath; // Path to uploaded image

    @Column(updatable = false)
    @com.fasterxml.jackson.annotation.JsonFormat(shape = com.fasterxml.jackson.annotation.JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}
